packages/api/src/command/medical/patient/get-patient.ts

```diff
@@ -151,3 +159,75 @@ export const getPatientWithDependencies = async ({
   const organization = await getOrganizationOrFail({ cxId });
   return { patient, facilities, organization };
 };
+
+/**
+ * This function calculates the similarity between two patients using the Jaro-Winkler algorithm.
+ * It returns a score between 0 and 1, where 1 means the patients are identical. We calculate scores
+ * for the following fields: First Name, Last Name, Address Line 1, Address Line 2, City, State, Country, Zipcode.
+ * This function won't be called if gender and DOB are not identical, so that is a givem.
+ * @param patient1
+ * @param patient2
+ * @returns The average of the similarity scores for each field.
+ */
+export const calculatePatientSimilarity = (
+  patient1: PatientData,
+  patient2: PatientData
+): number => {
+  let score = 0;
+  let fieldCount = 0;
+  const similarityScores: { [key: string]: [number, string?, string?] } = {};
+
+  const addScore = (field: string, value1: any, value2: any) => {
+    const similarity = jaroWinkler(value1, value2);
+    similarityScores[field] = [similarity, value1, value2];
+    score += similarity;
+    fieldCount += 1;
+  };
+
+  addScore("First Name", patient1.firstName, patient2.firstName);
+  addScore("Last Name", patient1.lastName, patient2.lastName);
+
+  // Calculate similarity for addresses
+  const address1 = patient1.address && patient1.address.length > 0 ? patient1.address[0] : null;
+  const address2 = patient2.address && patient2.address.length > 0 ? patient2.address[0] : null;
+
+  if (address1 && address2) {
+    if (address1.addressLine1 && address2.addressLine1) {
+      addScore("Address Line 1", address1.addressLine1, address2.addressLine1);
+    }
```

Comment:
What if addr1 has addressLine1 and addr2 doesn't? Shouldn't this decrease the score? Maybe this could be solved by having the fieldCount being calculated on `max(fieldCountPatient1, fieldCoundPatient2)` instead of increased on each call to `addScore()`?

Reply:
Hmmm not necessarily? I sort of thought about this as measuring the similarity amongst information entries that are different. So like if you have two patients and one has 4 datapoints and one has 5, in order to compare them and see if they are the same, you are going to look at the 4 datapoints that they share. Each datapoint is now weighted more heavily too. But the 5th one that only one had just isn't helpful to determine whether or not they are the same, since a missing data point doesn't say anything about the patient. Wdyt?

Reply:
Yeah, that make sense! 👍